Userland/Services/WebContent/WebDriverConnection.cpp

```diff
@@ -1807,6 +1813,43 @@ ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::handle_any_user_prompt
     return {};
 }
 
+// https://w3c.github.io/webdriver/#dfn-waiting-for-the-navigation-to-complete
+ErrorOr<void, Web::WebDriver::Error> WebDriverConnection::wait_for_navigation_to_complete()
+{
+    // 1. If the current session has a page loading strategy of none, return success with data null.
+    if (m_page_load_strategy == Web::WebDriver::PageLoadStrategy::None)
+        return {};
+
+    // 2. If the current browsing context is no longer open, return success with data null.
+    TRY(ensure_open_top_level_browsing_context());
+
+    // FIXME: 3. Start a timer. If this algorithm has not completed before timer reaches the session’s session page load timeout in milliseconds, return an error with error code timeout.
+    // FIXME: 4. If there is an ongoing attempt to navigate the current browsing context that has not yet matured, wait for navigation to mature.
+
+    // 5. Let readiness target be the document readiness state associated with the current session’s page loading strategy, which can be found in the table of page load strategies.
+    auto readiness_target = [this]() -> StringView {
+        switch (m_page_load_strategy) {
+        case Web::WebDriver::PageLoadStrategy::Normal:
+            return "complete"sv;
+        case Web::WebDriver::PageLoadStrategy::Eager:
+            return "interactive"sv;
+        default:
+            VERIFY_NOT_REACHED();
+        };
```

Comment:
This is rather unfortunate, especially since `Document::ready_state` will allocate in the loop below each time. You don't have to do it now, but we should add a raw accessor the the ready_state enum to Document, then this can be done without strings.